src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadataProvider.cs

```diff
@@ -232,15 +252,22 @@ protected virtual DefaultMetadataDetails[] CreatePropertyDetails(ModelMetadataId
         /// </remarks>
         protected virtual DefaultMetadataDetails CreateTypeDetails(ModelMetadataIdentity key)
         {
-            return new DefaultMetadataDetails(key, ModelAttributes.GetAttributesForType(key.ModelType));
+            ModelAttributes attributes;
+
+            if (key.MetadataKind == ModelMetadataKind.Parameter)
+            {
+                attributes = ModelAttributes.GetAttributesForParameter(key);
+            }
+            else
+            {
+                attributes = ModelAttributes.GetAttributesForType(key.ModelType);
+            }
+
+            return new DefaultMetadataDetails(key, attributes);
         }
```

Comment:
Should there be a new method here? These aren't really type details. I don't recall exactly the data flow here.

Reply:
Correct. This was definitely wrong. I think that was hanging over from my initial hacky spike. Made a new method for the parameter case.